src/python/aqueduct_executor/operators/utils/utils.py

```diff
@@ -88,12 +104,14 @@ def _read_json_input(storage: Storage, path: str) -> Any:
 
 def write_artifacts(
     storage: Storage,
+    artifact_types: List[OutputArtifactType],
     output_paths: List[str],
     output_metadata_paths: List[str],
     contents: List[Any],
     system_metadata: Dict[str, str],
-    artifact_types: List[OutputArtifactType],
+    user_defined_metadata: Dict[str, str],
```

Comment:
Created https://linear.app/aqueducthq/issue/ENG-1290/use-pydantic-object-for-artifact-metadata-in-backend-instead-of. I don't think this is in scope of this PR, since we don't the metadata path to pass information around anymore.